AK/Types.h

```diff
@@ -128,11 +128,34 @@ enum MemoryOrder {
     memory_order_seq_cst = __ATOMIC_SEQ_CST
 };
 
+template<class T>
```

Comment:
I don't think Types.h is the right place for this, if it's needed in the forwarding header, let's just put this in its own header and include it in Forward.h.

Also please put newlines between the methods in both these structs.

What does declaring, but not defining, the copy constructor achieve? I'm surprised the compiler didn't complain about the unused `d` variable.

Reply:
Nice catch on the copy ctors. They are meant to be empty function bodies. I goofed. I guess they are not being called anywhere so the linker doesn't actually need to find them? I will fix.